### PR TITLE
feat: [MLIR] Add basic fixed array support

### PR DESCRIPTION
**Pull request by Thirumalai-Shaktivel**
Towards: https://github.com/lfortran/lfortran/issues/4319

### CODE REVIEW

**Comment by Thirumalai-Shaktivel** [comment]
I wanted to enable a test, but it requires handling ArrayItem. I will add the required changes soon.

File: src/libasr/codegen/asr_to_mlir.cpp [hunk 1]
```diff
@@ -65,13 +65,51 @@ class ASRToMLIRVisitor : public ASR::BaseVisitor<ASRToMLIRVisitor>
...
+            } default: {
+                throw LCompilersException("Variable type '"+
+                    ASRUtils::type_to_str_python(asr_type) +
+                    "' is not supported yet");
+            }
+        }
+    }
+
+    std::string getUniqueName(std::string name = "") {
+        static int itr = 0; ++itr;
```
Comment (Thirumalai-Shaktivel) [inline src/libasr/codegen/asr_to_mlir.cpp:91]: Same, let me know, if this is okay.

**Comment by Thirumalai-Shaktivel** [comment]
If the tests pass, this PR is ready for review!

File: src/libasr/codegen/asr_to_mlir.cpp (hunk 1 above)
Comment (certik) [reply src/libasr/codegen/asr_to_mlir.cpp:97]: For now it's fine. But you should use the `_` style of naming.

**Review by certik — APPROVED** [review]

File: src/libasr/codegen/asr_to_mlir.cpp (hunk 1 above)
Comment (Thirumalai-Shaktivel) [reply src/libasr/codegen/asr_to_mlir.cpp:97]: Yes. I will them in the subsequent PRs.

**Comment by Thirumalai-Shaktivel** [comment]
Thanks for the review!

**Pull request merged**